### PR TITLE
[AWS Orgs] Add cloud organization fields

### PR DESCRIPTION
**Pull request by orestisfl**
### Screenshot/Data
image

### Related Issues

- Fixes: #1177 

### Checklist
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added the necessary README/documentation (if appropriate)

### CODE REVIEW

File: resources/providers/awslib/account_provider.go [hunk 1]
```diff
@@ -53,8 +71,11 @@ func listAccounts(ctx context.Context, client organizations.ListAccountsAPIClien
 			}
 
 			accounts = append(accounts, cloud.Identity{
-				Account:      *account.Id,
-				AccountAlias: strings.Dereference(account.Name),
+				Provider:         "aws",
+				Account:          *account.Id,
```
Comment (uri-weisman) [inline resources/providers/awslib/account_provider.go:75]:
```suggestion
				Account:          strings.Dereference(account.Id),
```

File: dataprovider/providers/cloud/data_provider.go [hunk 2]
```diff
@@ -32,12 +34,17 @@ const (
 	cloudAccountNameField = "cloud.account.name"
 	cloudProviderField    = "cloud.provider"
 	cloudRegionField      = "cloud.region"
+	// TODO: update fields names when an ECS field is decided
+	cloudOrganizationIdField   = "cloud.Organization.id"
+	cloudOrganizationNameField = "cloud.Organization.name"
```
Comment (uri-weisman) [inline dataprovider/providers/cloud/data_provider.go:39]:
```suggestion
	cloudOrganizationIdField   = "cloud.organization.id"
	cloudOrganizationNameField = "cloud.organization.name"
```

File: dataprovider/providers/cloud/data_provider.go [hunk 3]
```diff
@@ -66,27 +73,14 @@ func (a DataProvider) FetchData(_ string, id string) (types.Data, error) {
...
-
-	return nil
+	return errors.Join(
+		insertIfNotEmpty(cloudAccountIdField, strings.FirstNonEmpty(resMetadata.AwsAccountId, a.accountId), event),
+		insertIfNotEmpty(cloudAccountNameField, strings.FirstNonEmpty(resMetadata.AwsAccountAlias, a.accountName), event),
+		insertIfNotEmpty(cloudProviderField, a.providerName, event),
+		insertIfNotEmpty(cloudRegionField, resMetadata.Region, event),
+		insertIfNotEmpty(cloudOrganizationIdField, resMetadata.AwsOrganizationId, event),
+		insertIfNotEmpty(cloudOrganizationNameField, resMetadata.AwsOrganizationName, event),
+	)
```
Comment (uri-weisman) [inline dataprovider/providers/cloud/data_provider.go:83]: 👍

**Review by uri-weisman — CHANGES_REQUESTED** [review]

File: resources/providers/awslib/account_provider.go (hunk 1 above)
Comment (orestisfl) [reply resources/providers/awslib/account_provider.go:75]: `account.Id` is already checked for

File: dataprovider/providers/cloud/data_provider.go (hunk 2 above)
Comment (orestisfl) [reply dataprovider/providers/cloud/data_provider.go:39]: Capitalized as non-final since they don't exist in ECS yet

Comment (uri-weisman) [reply dataprovider/providers/cloud/data_provider.go:39]: Makes sense, however, one of the reasons for adding those fields is to help users perform insightful queries and they would expect an ECS format.

File: dataprovider/providers/cloud/data_provider.go [hunk 4]
```diff
@@ -32,12 +34,17 @@ const (
 	cloudAccountNameField = "cloud.account.name"
 	cloudProviderField    = "cloud.provider"
 	cloudRegionField      = "cloud.region"
+	// TODO: update fields names when an ECS field is decided
```
Comment (oren-zohar) [inline dataprovider/providers/cloud/data_provider.go:37]: can we track it somehow? maybe add a link to this part in the ticket about the ECS?

File: resources/providers/awslib/account_provider.go [hunk 5]
```diff
@@ -65,3 +82,63 @@ func listAccounts(ctx context.Context, client organizations.ListAccountsAPIClien
...
+	for paginator.HasMorePages() {
+		o, err := paginator.NextPage(ctx)
+		if err != nil {
+			return organizationalUnitInfo{}, err
+		}
+		if len(o.Parents) == 0 {
+			continue
+		}
+
+		parent := o.Parents[0]
```
Comment (oren-zohar) [inline resources/providers/awslib/account_provider.go:102]: might worth adding some context on why we're picking the first one

**Review by oren-zohar — APPROVED** [review]

File: resources/providers/awslib/account_provider.go (hunk 5 above)
Comment (orestisfl) [reply resources/providers/awslib/account_provider.go:102]: Added a few comments

File: dataprovider/providers/cloud/data_provider.go (hunk 4 above)
Comment (orestisfl) [reply dataprovider/providers/cloud/data_provider.go:37]: Is there such a ticket?

**Pull request merged**